### PR TITLE
Phase2-hg302 Bug fix for the corners and replacing cout with message logger services

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Bug fix for the corners and replacing cout with message logger services. The corner calculation for the rotated layers was wrong. The  corner calculation is useful for firework studies 

#### PR validation:

Use the runTheMatrix test workflows 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special - may need to bckport the bug fixes in HGCal geometry

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by perrotta** [comment]
@bsunanda could you please quickly summarize in the PR description which is the bug that you aim fixing with this PR?

**Comment by bsunanda** [comment]
@cvuosalo @srimanob Please sign this PR. It is worthwhile to backport the bug fixing part to 12_3_X

**Comment by srimanob** [comment]
@cmsbuild please test

This is to wipe out unexpected failure comparison.

**Comment by srimanob** [comment]
Hi @bsunanda 
As you mentioned in the PR description,` The corner calculation for the rotated layers was wrong.`, do we expect to see change(s) in PR test result, or what do you expect to see? Thanks for clarification.

In the future, I would suggest not to couple between bug-fix and technical PR (i.e. message logger replacement) together. This decouple PRs will keep a clear between bug-fix PR and technical PR.

**Comment by srimanob** [comment]
type bug-fix

**Comment by bsunanda** [comment]
When there is a bug fix, it is expected to change the predictions.

**Comment by perrotta** [comment]
ping bot

**Comment by perrotta** [comment]
> When there is a bug fix, it is expected to change the predictions.

@bsunanda it would help (and speed up) the review process if, when a bug is reported and a PR to fix it is submitted, you list:
- where is that bug, and how did you fix it
- what was the effect of that bug
- what is the difference in outputs expected after the fix

Only saying that "when there is a bug fix, it is expected to change the predictions" does not help in judging whether the effects of the fix are as wanted, or if there are any unplanned and unwanted side effects instead.

**Comment by bsunanda** [comment]
The bug was in the treatment of rotated layers. In computing the corner coordinates, it first retrieves the cell position in the local frame of the wafer, then computes the corner coordinates in that local frame and finally apply the transformation to global frame. In the earlier version computation of the cell position in the local frame had an error (some rotation was applied for rotated layers). Clearly this used to lead to wrong results. Now this is avoided by inhibiting rotation for this computation. The implication of this on the output should be there, but cannot judge how much it will be

**Comment by cvuosalo** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

This is a bug fix PR for the  rotated layers of HGCAL. We need to follow up from the validation result from the next round of Phase-2 validation.
FYI @cms-sw/pdmv-l2

File: Geometry/HGCalGeometry/interface/HGCalGeometry.h [hunk 1]
```diff
@@ -71,6 +71,7 @@ class HGCalGeometry final : public CaloSubdetectorGeometry {
                   CaloSubdetectorGeometry::IVec& dinsVector) const override;
 
   GlobalPoint getPosition(const DetId& id) const;
```
Comment (perrotta) [inline Geometry/HGCalGeometry/interface/HGCalGeometry.h:73]: having only one `getPosition` method with the `bool debug = false` as default would have done the same without duplication

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
+1
- Effect of the bug are not visible in the bot test outputs: they would be probably observed in a larger scale validation campaign

**Pull request merged**